README.md

```diff
@@ -1,25 +1,223 @@
+==================
 
-Allows to manage zabbix hosts
+- Website: https://www.terraform.io
+- [![Gitter chat](https://badges.gitter.im/hashicorp-terraform/Lobby.png)](https://gitter.im/hashicorp-terraform/Lobby)
+- Mailing list: [Google Groups](http://groups.google.com/group/terraform-tool)
 
+<img src="https://cdn.rawgit.com/hashicorp/terraform-website/master/content/source/assets/images/logo-hashicorp.svg" width="600px">
+
+Requirements
+------------
+
+-	[Terraform](https://www.terraform.io/downloads.html) 0.12.x
+-	[Go](https://golang.org/doc/install) 1.12 (to build the provider plugin)
+
+Building The Provider
+---------------------
+
+Clone repository to: `$GOPATH/src/github.com/claranet/terraform-provider-zabbix`
+
+```sh
+$ mkdir -p $GOPATH/src/github.com/claranet; cd $GOPATH/src/github.com/claranet
+$ git clone git@github.com:claranet/terraform-provider-zabbix
+```
+
+Enter the provider directory and build the provider
+
+```sh
+$ cd $GOPATH/src/github.com/claranet/terraform-provider-zabbix
+$ make build
+```
+
+**Note**: For contributions created from forks, the repository should still be cloned under the `$GOPATH/src/github.com/claranet/terraform-provider-zabbix` directory to allow the provided `make` commands to properly run, build, and test this project.
+
+Using the provider
+------------------
+If you're building the provider, follow the instructions to [install it as a plugin.](https://www.terraform.io/docs/plugins/basics.html#installing-a-plugin) After placing it into your plugins directory,  run `terraform init` to initialize it.
+
+Further [usage documentation is available on the Terraform website](https://www.terraform.io/docs/providers/zabbix/index.html).
+
+Developing the Provider
+-----------------------
+
+If you wish to work on the provider, you'll first need [Go](http://www.golang.org) installed on your machine (version 1.11+ is *required*). You'll also need to correctly setup a [GOPATH](http://golang.org/doc/code.html#GOPATH), as well as adding `$GOPATH/bin` to your `$PATH`.
+
+To compile the provider, run `make build`. This will build the provider and put the provider binary in the `$GOPATH/bin` directory.
+
+```sh
+$ make build
+...
+$ $GOPATH/bin/terraform-provider-zabbix
+...
+```
+
+In order to test the provider, you can simply run `make test`.
+
+```sh
+$ make test
 ```
+
+In order to run the full suite of Acceptance tests, run `make testacc`.
+
+*Note:* Acceptance tests create real resources, and often cost money to run.
+
+```sh
+$ make testacc
+```
+
+Notes
+-----
+
+### Template linking
+
+Template link is used as a way to track template items and triggers. Template link must contain all the id ofs your local items and triggers otherwise they will be delete during the next apply.
+
+* If you use the template link resource to track template depencencies you should pay attention to always have you local item and trigger declared inside otherwise they will be delete and create in loop.
```

Comment:
```suggestion
* If you use the template link resource to track template dependencies you should pay attention to always have your local item and trigger referenced otherwise they will be deleted and will need to be recreated, causing a permanent diff.
```